### DISCUSSION TITLE
HOW TO USE pixijs Create sprites based on pixels

### DISCUSSION THREAD

**Discussion by a6965921** [pixijs/pixijs#8002] [Q&A] [answered]
`let d = new PIXI.Sprite('#F00’)`

**Comment by jramstedt** [comment]
You can draw filled rectangle with https://pixijs.download/dev/docs/PIXI.Graphics.html

**Comment by a6965921** [comment]
> You can draw filled rectangle with https://pixijs.download/dev/docs/PIXI.Graphics.html

i want add to particlecontainer

**Comment by JackyVerse** [comment]
> > You can draw filled rectangle with https://pixijs.download/dev/docs/PIXI.Graphics.html
> 
> i want add to particlecontainer


Try this one 
var sprites = new PIXI.particles.ParticleContainer(10000, {
    scale: true,
    position: true,
    rotation: true,
    uvs: true,
    alpha: true
});

**Comment by bigtimebuddy** [comment]
If you need help getting started, please use discussions instead of creating an issue. 

Also, seems like you're just getting started with Pixi (congrats!), consider checking out these resources first:

* [PixiJS Guides](https://pixijs.io/guides)
* [PixiJS Elementals](https://www.pixijselementals.com/#introduction)
* [PixiJS Examples](https://pixijs.io/examples)

**Comment by eXponenta** [comment, marked as answer]
If you want has billion sprites, you can use a trick:
```
const sprite = new PIXI.Sprite(PIXI.Texture.WHITE);
sprite.tint = 0xff0000;
```

Or can use  a canvas as color source in texture with 1x1 dimensions:

```
const canvas = document.createElement('canvas');
canvas.width = canvas.height = 256;
const ctx = canvas.getContext('2d');
....
const base = PIXI.BaseTexture.from(canvas);
let pixelIndex = 0;
let store = [];

function createPixel(color) {
     if (!store[color]) {
           pixelIndex ++;
           const x = pixelIndex % canvas.width;
           const y = pixelIndex / canvas.height | 0;
           store[color] = new PIXI.Texture(base, new PIXI.Rectangle(x + 0.25, y + 0.25, 0.5, 0.5));
           ctx.fill= color;
           ctx.fillRect(x, y, 1, 1);
            base.update()
     }
    
     return new PIXI.Sprite(store[color])
}

```

Or must use a direct buffers usage for 2000-10k dots

https://www.pixiplayground.com/#/edit/5bWjCjL70izwdVOncfzoM
